### DISCUSSION TITLE
Posibility to load x86 shared libraries on arm64 apps

### DISCUSSION THREAD

**Discussion by Javier97sm** [ptitSeb/box64#1378] [Q&A]
I would like to run a native application on an arm64 device, with the particularity that I am interested in using box64 to load shared libraries that are compiled in x86. I have tried just to make it work but for what I have seen (I might be wrong) if a native application is launched and tries to load a x86 shared object, box64 will not be called.

For now I have though of a workaround but it will be a lot of work:

Instead of running the native application (which I have the code), I can make a main() in x86 that calls the real application as a shared library in arm64, so when the real x86 library I want to launch gets called box64 will do its emulation magic, the problem with this solution is that i would have to wrap tons of symbols of the native application (now transformed in library)...

Does anyone know any possible way of archiving this?

**Comment by ksco** [comment]
Related: https://github.com/ptitSeb/box64/issues/1273 https://github.com/ptitSeb/box64/pull/1362
